lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -415,4 +418,31 @@
         return A;
     });
 
+    platform.registerFunction(FunctionsEnum.ObjectFromEntries, function (iterable) {
+        // #sec-object.fromentries
+        "use strict";
+        if (this === null || this === undefined) {
+            __chakraLibrary.raiseNeedObject("Object.prototype.fromEntries");
+        }
+
+        const o = {};
+        const propDescriptor = {
+            enumerable : true,
+            configurable : true,
+            writable : true,
+            value : undefined
+        };
+
+        let key;
+        for (const entry of iterable) {
+            if (typeof entry !== "object") {
+                __chakraLibrary.raiseNeedObject("Object.prototype.fromEntries");
```

Comment:
This isn't a great error message but not sure what I should use.

Reply:
I think this isn't required; if I understand the spec correctly, then this part of the implementation is an inlined version of `CreateDataPropertyOnObject`, and the `this` of that function is the `o` here which is guaranteed to be an object

Reply:
Oh, I think I'm mistaken; this is corresponding to the internals of `AddEntriesFromIterable`, which does require that the iterable returns objects. Nevermind then!

Reply:
s/Object.prototype.fromEntries/Object.fromEntries/